src/ansible_builder/user_definition.py

```diff
@@ -243,3 +243,13 @@ def validate(self):
                     self.build_arg_defaults['EE_BUILDER_IMAGE'] = self.builder_image.name
 
             self._validate_additional_build_files()
+
+            if self.version == 3:
+                build_steps = self.raw.get('additional_build_steps', {})
+                for step_name, steps in build_steps.items():
+                    for directive in steps:
+                        if directive.startswith('USER '):
+                            raise DefinitionError(
```

Comment:
I'm not certain that a hard error is the thing to do. As @nitzmahone suggested in the issue, we probably just want to output a warning message instead. Could be a valid reason for someone wanting to change the user.